nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/test/java/org/nuxeo/ecm/platform/ec/notification/TestEmailNotification.java

```diff
@@ -0,0 +1,182 @@
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.Date;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.common.utils.StringUtils;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.NuxeoPrincipal;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventService;
+import org.nuxeo.ecm.core.event.impl.DocumentEventContext;
+import org.nuxeo.ecm.platform.notification.api.NotificationManager;
+import org.nuxeo.ecm.platform.test.PlatformFeature;
+import org.nuxeo.mail.SmtpMailServerFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * Tests the whole email notification process: from subscribing to a given event to the reception of the email for this
+ * event.
+ *
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ PlatformFeature.class, SmtpMailServerFeature.class })
+@Deploy("org.nuxeo.ecm.platform.notification.core")
+@Deploy("org.nuxeo.ecm.platform.notification.api")
+@Deploy("org.nuxeo.ecm.platform.url.api")
+@Deploy("org.nuxeo.ecm.platform.url.core")
+@Deploy("org.nuxeo.ecm.platform.notification.core.tests:OSGI-INF/notification-event-listener-contrib.xml")
+public class TestEmailNotification {
+
+    protected static final String DUMMY_NOTIFICATION_NAME = "DummyNotificationToSendMail";
+
+    protected static final String DUMMY_EVENT_NAME = "dummyNotificationToSendMail";
+
+    protected static final String DOCUMENT_NAME = "anyFile";
+
+    protected static final String ANOTHER_DOCUMENT_NAME = "anyFile2";
+
+    protected static final SimpleDateFormat EVENT_DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy - HH:mm");
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected NotificationManager notificationManager;
+
+    @Inject
+    protected EventService eventService;
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Inject
+    protected SmtpMailServerFeature.MailsResult emailsResult;
+
+    protected DocumentModel domain;
+
+    @Before
+    public void before() {
+        domain = session.createDocumentModel("/", "domain", "Domain");
+        domain = session.createDocument(domain);
+    }
+
+    @Test
+    public void shouldReceiveNotificationMailWhenSubscribeToDocument() {
+        // create a File document
+        DocumentModel documentModel = createDocument(domain, DOCUMENT_NAME);
+
+        // subscribe to the dummy notification
+        addSubscription(documentModel);
```

Comment:
Renaming the method addDummySubscription would be more explicit, and the comment could be removed.